modules/objdetect/include/opencv2/objdetect.hpp

```diff
@@ -763,10 +763,53 @@ class CV_EXPORTS_W QRCodeEncoder {
 
 };
 
+struct CV_EXPORTS_W_SIMPLE QRDetectorParameters {
+
+    /** @brief Methods for detecting QR codes: CONTOURS_BASED, ARUCO_BASED */
+    enum Method {
+        CONTOURS_BASED, ///< QR detect method with detect finder pattern as contours
+        ARUCO_BASED ///< QR detect method, with detect finder pattern as ArUco marker
+        // TODO: add WECHAT method
+    };
+
+    QRDetectorParameters();
+
+    CV_PROP_RW Method qrDetectMethod;
+
+    /** @brief The minimum allowed size of a QR module in the smallest image in the image pyramid. */
+    CV_PROP_RW float minModuleSizeInPyramid = 4.f;
+
+    /** @brief The maximum allowed relative rotation for finder patterns in the same QR code. */
+    CV_PROP_RW float maxRotation = (float)CV_PI/12.f;
+
+    /** @brief The maximum allowed relative mismatch in module sizes for finder patterns in the same QR code. */
+    CV_PROP_RW float maxModuleSizeMismatch = 1.75f;
+
+    /** @brief The maximum allowed module relative mismatch for timing pattern module
+     *
+     * If relative mismatch of timing pattern module more this values, calculateTimingPattern() will add penalty points.
+     * If calculateTimingPattern() adds a lot of penalty points, QR code will be rejected. */
+    CV_PROP_RW float maxTimingPatternMismatch = 2.f;
+
+    /** @brief The maximum allowed percentage of penalty points out of total pins in timing pattern. */
+    CV_PROP_RW float maxPenalties = 0.4f;
+
+    /** @brief The maximum allowed relative color mismatch in the timing pattern. */
+    CV_PROP_RW float maxColorsMismatch = 0.2f;
+
+    /** @brief The algorithm find QR codes with almost minimum timing pattern score and minimum size
+     *
+     * The QR code with the minimum "timing pattern score" and minimum "size" is selected as the best QR code.
+     * If for the current QR code "timing pattern score" * scaleTimingPatternScore < "previous timing pattern score" and "size" < "previous size", then
+     * current QR code set as the best QR code */
+    CV_PROP_RW float scaleTimingPatternScore = 0.9f;
```

Comment:
Please add note to all params, if they related to `ARUCO_BASED` algorithm, `CONTOURS_BASED` algorithm or both.

Reply:
added info